### ISSUE TITLE
[FoxyTab][Feature request] Option for "close duplicate" feature

### ISSUE THREAD

**Issue by jerome-dl** [erosman/support#651]
Hi team,

I would be happy if the _Close duplicate_ feature could take into consideration **containers**.
As it can be legitimate to keep 2 open tabs with same URL in different containers, but not 2 open tabs with same URL and same container.

Possibly, _Close duplicate_ feature may present a list of duplicate tabs, where we could select which tabs to be closed (or not).

Thanks,
Jérôme

**Comment by erosman** [comment]
It is possible but there are many options which may not be practical to include them all e.g.

- include/exclude containers
- include/exclude each containers separately
- include/exclude pinned tabs
- include/exclude discarded tabs
- include/exclude blank pages (already requested in another issue)
- ...

**Comment by erosman** [comment]
> [!Note]
> 
> I will be rewriting all my extensions for MV3 (manifest version 3) in due course, and ALL requests will be consider at the time.

**Comment by nisc** [comment]
Quick note (not 100% on-topic): I noticed the native "Close Duplicate Tabs" feature in Firefox recently. Did I miss it before, or is it new?

When I tested it briefly, it closed 1 tab less than the FoxyTab feature, but I didn't spend the time to debug how exactly it's different.

TL;DR: FoxyTab should probably clarify how its feature is different, if it is to stay.

**Comment by erosman** [comment]
> I noticed the native "Close Duplicate Tabs" feature in Firefox recently. Did I miss it before, or is it new?

Firefox has been adding FoxyTab's features for quite a while now. :)

All FoxyTab features where unique when FoxyTab first came out.
I made them because they were missing from the Firefox’s native menu e.g. Duplicate Tab, Discard (Unload) Tab, Move Tab (to start, to end, to new window), Close Duplicate Tabs, ....etc

**Comment by nisc** [comment]
> Firefox has been adding FoxyTab's features for quite a while now. :)

Sue them! ;-)

Okay, got it. But my comment regarding the inconsistent behavior still holds. Unfortunately, I didn't document the test case, but the two implementations didn't behave identically.
